optimizer/resolver.go

```diff
@@ -100,6 +100,8 @@ type resolverContext struct {
 	inCreateOrDropTable bool
 	// When visiting show statement.
 	inShow bool
+	// When visiting constraint foreign key.
+	inConstraintForeignKey bool
```

Comment:
When table does not exist, foreign key should fail to resolve, so this is not needed.